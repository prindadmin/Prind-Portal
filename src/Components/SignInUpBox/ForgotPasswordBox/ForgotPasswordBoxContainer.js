import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'

import * as authReducer from '../../../Reducers/authReducer'

import ForgotPasswordBox from './ForgotPasswordBox'

const mapStatetoProps = state => {
  return {
    auth: state.auth,
    user: state.user,
  }
}

const mapDispatchToProps = dispatch => {
  return {
    resetPassword: (userDetails, resolve, reject) => {
      dispatch(authReducer.forgotPassword(userDetails, resolve, reject))
    }
  }
}

export default withRouter(connect(mapStatetoProps, mapDispatchToProps)(ForgotPasswordBox))
