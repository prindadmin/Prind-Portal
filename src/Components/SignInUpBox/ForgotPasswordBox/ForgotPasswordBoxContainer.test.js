import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import {
  HashRouter as Router
} from "react-router-dom";

import * as Component from './ForgotPasswordBoxContainer'
//import * as States from '../../../States'

// https://www.robinwieruch.de/react-connected-component-test

const mockStore = configureStore([]);

var component;

beforeEach(() => {
  const store = mockStore({});

  const mockToggleVisibleForm = jest.fn()
  const mockResetPassword = jest.fn()
  const props = {
    toggleVisibleForm: mockToggleVisibleForm,
    resetPassword: mockResetPassword,
  }

  component = renderer.create(
    <Provider store={store}>
      <Router>
        <Component.default {...props} />
      </Router>
    </Provider>
  );
});


it('should render with given state', () => {
  expect(component.toJSON()).toMatchSnapshot();
});
