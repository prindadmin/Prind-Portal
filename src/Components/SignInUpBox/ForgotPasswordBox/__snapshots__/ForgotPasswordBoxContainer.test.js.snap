// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with given state 1`] = `
Array [
  <div
    className="logo-container"
  >
    <a
      href="https://prind.tech"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Prin-D Technology logo"
        src="/images/logos/prind-tech-logo.webp"
      />
    </a>
  </div>,
  <div
    className="welcome-text-heading"
  >
    Welcome
  </div>,
  <div
    className="welcome-text-body"
  >
    Please fill in your email address to request a password reset
  </div>,
  <form
    className="sign-in-form"
    id="forgot-password-form"
    onSubmit={[Function]}
  >
    <input
      className="default"
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value=""
    />
    <div
      className="spacer"
    />
    <input
      className="submit-button"
      id="submit"
      onClick={[Function]}
      type="submit"
      value="Reset password"
    />
    <div
      className="spacer"
    />
    <p
      className="sign-up-in-text"
      id="sign-in-form-text"
      onClick={[Function]}
    >
      Already have an account?  Click here to sign in
    </p>
  </form>,
]
`;
